python/lsst/sims/maf/slicers/oneDSlicer.py

```diff
@@ -110,19 +110,21 @@ def _sliceSimData(islice):
 
     def __eq__(self, otherSlicer):
         """Evaluate if slicers are equivalent."""
-        match = False
+        result = False
         if isinstance(otherSlicer, OneDSlicer):
             if self.sliceColName == otherSlicer.sliceColName:
                 # If slicer restored from disk or setup, then 'bins' in slicePoints dict.
+                # This is preferred method to see if slicers are equal.
                 if ('bins' in self.slicePoints) & ('bins' in otherSlicer.slicePoints):
-                    match = np.all(otherSlicer.slicePoints['bins'] == self.slicePoints['bins'])
-                # Otherwise, try some other things.
+                    result = np.all(otherSlicer.slicePoints['bins'] == self.slicePoints['bins'])
+                # However, even before we 'setup' the slicer with data, the slicers could be equivalent.
                 else:
-                    if self.bins & otherSlicer.bins: # if these are not None
-                        match = np.all(self.bins == otherSlicer.bins)
-                    elif self.binsize & self.binMin & self.binMax &\
-                        otherSlicer.binsize & otherSlicer.binMin & otherSlicer.binMax:
-                        if (self.binsize == otherSlicer.binsize) &\
-                          (self.binMin == otherSlicer.binMin) & (self.binMax == otherSlicer.binMax):
-                            match = True
-        return match
+                    if (self.bins is not None) and (otherSlicer.bins is not None):
```

Comment:
Hmm, apparently it actually is important. If it's not None, it's a numpy array. 
The numpy array can't be compared with "and", but & fails if it's None.